File: python/examples/roundtrip.py
#!/usr/bin/env python3
"""
Round Trip Example

Complete trade cycle: buy then sell to end up flat.

Requirements:
    pip install hyperliquid-sdk

Usage:
    export PRIVATE_KEY="0x..."
    python roundtrip.py
"""

import os
import sys

from hyperliquid_sdk import HyperliquidSDK

PRIVATE_KEY = os.environ.get("PRIVATE_KEY")

if not PRIVATE_KEY:
    print("Error: Set PRIVATE_KEY environment variable")
    print("  export PRIVATE_KEY='0xYourPrivateKey'")
    sys.exit(1)


def main():
    print("Hyperliquid Round Trip Example")
    print("=" * 50)

    sdk = HyperliquidSDK(private_key=PRIVATE_KEY)
    print(f"Address: {sdk.address}")

    # Buy $11 worth of BTC
    print("Buying BTC...")
    buy = sdk.market_buy("BTC", notional=11)
    filled_size = buy.filled_size or buy.size
    print(f"  Bought: {filled_size} BTC")
    print(f"  Status: {buy.status}")

    # Sell the same amount
    print("Selling BTC...")
    sell = sdk.market_sell("BTC", size=filled_size)
    print(f"  Sold: {sell.filled_size or sell.size} BTC")
    print(f"  Status: {sell.status}")

    print("Done! Position should be flat.")


if __name__ == "__main__":
    main()

File: typescript/examples/roundtrip.ts
#!/usr/bin/env npx ts-node
// @ts-nocheck
/**
 * Round Trip Example
 *
 * Complete trade cycle: buy then sell to end up flat.
 */

import { HyperliquidSDK } from '@quicknode/hyperliquid-sdk';

const PRIVATE_KEY = process.env.PRIVATE_KEY;

if (!PRIVATE_KEY) {
  console.error("Set PRIVATE_KEY environment variable");
  process.exit(1);
}

async function main() {
  const sdk = new HyperliquidSDK(undefined, { privateKey: PRIVATE_KEY });

  // Buy $11 worth of BTC
  console.log("Buying BTC...");
  const buy = await sdk.marketBuy("BTC", { notional: 11 });
  console.log(`  Bought: ${buy.filledSize || buy.size} BTC`);
  console.log(`  Status: ${buy.status}`);

  // Sell the same amount
  console.log("Selling BTC...");
  const sell = await sdk.marketSell("BTC", { size: buy.filledSize || buy.size });
  console.log(`  Sold: ${sell.filledSize || sell.size} BTC`);
  console.log(`  Status: ${sell.status}`);

  console.log("Done! Position should be flat.");
}

main().catch(console.error);
